### DISCUSSION TITLE
Is there a way to use Alphatab with Electron?

### DISCUSSION THREAD

**Discussion by chz8494** [CoderLine/alphaTab#1099] [General]
I've developed a website which uses Alphatab to show scores, now I'm trying to make a desktop version of it and may add on more features. But when I try to load Alphatab API via `go-astilectron` it says 
```
alphaTab.js:141 Uncaught Error: Usage of AlphaTabApi is only possible in browser environments. For usage in node use the Low Level APIs
```
After spending some time reading https://alphatab.net/docs/guides/nodejs/ and going through some other similar issues, feels like I have to give up using `alphaTab.AlphaTabApi(main, settings)` but call this?
```
const score = alphaTab.importer.ScoreLoader.loadScoreFromBytes(
  new Uint8Array(fileData),
  settings
);
settings.core.engine = "svg";
const renderer = new alphaTab.rendering.ScoreRenderer(settings);
...
renderer.renderScore(score, [0]);
```
and I won't be able to use any API features that `AlphaTabApi` provides such as `api.scoreLoaded`, `api.render()`,`api.playPause()`...etc?
Are there better way? or more guide on how to make it function similar to your demo on https://alphatab.net/docs/tutorial-web/introduction ?

**Comment by Danielku15** [comment]
Generally there should not be any restriction to use it in Electron. 

What is important in such setups is whether alphaTab is going to be used in the backend (quasi server) part or on the frontend part. If you use it on the backend part inside NodeJs you cannot use `AlphaTabApi` as it is targeting the browser as UI on the backend you would use the [Low Level APIs](https://alphatab.net/docs/guides/lowlevel-apis) to load files, and render them to SVG. There is an open backlog item to have a NodeJS level rendering using abstractions like skia-canvas. Until now you would be limited to generating SVG strings. 

If you plan to use AlphaTab in the frontend part I wouldn't expect any issues unless some "browser" features are disabled by setting.  
I haven't dived into any details of the latest Electron features or how `go-astilectron` is organized. When I used it the last time you had a backend/server part where you can operate APIs and access some system level resources. And then you have basically a browser where you can open any HTML which then uses any APIs available to display a native looking app. 

In most cases you would like to use alphaTab as UI control in the frontend of the Electron app. You could also offload things to the "backend" inside APIs and use alphaTab there too, but that might be an edge case.
